submissions/nataliaArtsiukh/Frogger-game/app.js

```diff
@@ -0,0 +1,141 @@
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+}
+
+// Enemies our player must avoid
+
+var Enemy = function(numRow) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.x = colWidth * -1;
+    this.y = rowHeight * numRow - 20;
+    this.speed = (Math.random() + 0.7) * colWidth;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    if (this.x < colWidth * 5) {
+        this.x = this.x + this.speed * dt;
+    } else {
+        this.x = colWidth * -1;
+        this.speed = (Math.random() + 0.7) * colWidth;
+    }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+var Player = function() {
+    this.x = colWidth * 2;
+    this.y = rowHeight * 5 - 10;
+    this.finished = false;
+    this.sprite = 'images/char-boy.png';
+    this.update = function(){
+        for (var i = 0; i < allEnemies.length; i++) {
+            if (this.y == allEnemies[i].y + 10) {
+                if (allEnemies[i].x > this.x - colWidth + 20 && allEnemies[i].x < this.x + colWidth - 20) {
+                    this.x = colWidth * 2;
+                    this.y = rowHeight * 5 - 10;
+                } 
+            }
+        }
+        if (this.y < 0) {
+            if (!this.finished) {
+                setTimeout(() => {
+                    this.x = colWidth * 2;
+                    this.y = rowHeight * 5 - 10;
+                    this.finished = false;
+                }, 1000);
+                this.finished = true;
+            }
+        }
+    };
+    this.render = function(){
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    };
+    this.handleInput = function(direction) {
+        if (this.y > 0) {
+            if (direction == 'left' && this.x > 0) {
+                this.x -= colWidth;
+            } else if (direction == 'up' && this.y > 0) {
+                this.y -= rowHeight;
+            } else if (direction == 'right' && this.x < colWidth * 4) {
+                this.x += colWidth;
+            } else if (direction == 'down' && this.y < rowHeight * 5) {
+                this.y += rowHeight;
+            }
+        } 
+    };
+}
+
+// Now instantiate your objects.
+// Place all enemy objects in an array called allEnemies
+// Place the player object in a variable called player
+
+var allStars;
+var allEnemies;
+var player;
```

Comment:
Switch to `const` and `let` syntax.

Reply:
I wrote in the style of the original code, but now code is corrected)